src/main/kotlin/no/nav/syfo/KafkaModule.kt

```diff
@@ -121,22 +121,39 @@ suspend fun createListener(applicationState: ApplicationState, action: suspend C
     }
 
 @KtorExperimentalAPI
-suspend fun blockingApplicationLogic(
+suspend fun blockingApplicationLogicSmRegTopic(
     applicationState: ApplicationState,
     database: DatabaseInterface,
     env: Environment,
     kafkaConsumer: KafkaConsumer<String, String>
 ) {
     while (applicationState.ready) {
-        kafkaConsumer.poll(Duration.ofMillis(0)).forEach { consumerRecord ->
-            handleReceivedMessage(database, env, consumerRecord)
+        pollAndProcessSMRegTopic(kafkaConsumer, database, env)
+    }
+}
+
+suspend fun pollAndProcessSMRegTopic(
+    kafkaConsumer: KafkaConsumer<String, String>,
+    database: DatabaseInterface,
+    env: Environment
+) {
+    val starttime = System.currentTimeMillis()
+    val consumerRecords = kafkaConsumer.poll(Duration.ofMillis(1000))
```

Comment:
Hva skjer egentlig når man bytter ut `0` med `1000` her?

Reply:
Slik jeg oppfatter det: Da gjøres det et kall til Kakfa som returnerer senest etter 1000 ms, men det kan returnere før dersom Kafka klarer å returnere det antall meldinger som man har konfigurert. Så jeg synes det er bedre å bruke denne mekanismen i Kafka enn å bruke delay, for da blir det ingen venting så lenge Kafka har data til oss.

Reply:
Skjønner, det høres bra ut 👍

Reply:
Et annet moment som jeg ser når jeg tester lokalt: Hvis man bruker 0 som parameter så returnerer de fleste poll-kallene *ingen* records, så det blir mange ekstra roundtrips til Kafka uten at man kommer noe videre. Dette skjer selv om det er mange meldinger i topic'en.